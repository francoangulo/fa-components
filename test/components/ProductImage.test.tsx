import * as React from "react";
import { ProductImage } from "../../src/components";

import renderer from "react-test-renderer";
import { ProductCard } from "../../src/components/ProductCard";
import { product2 } from "../data/products";

describe("ProductImage", () => {
  test("Renders ProductImage with a custom image without crashing", () => {
    const wrapper = renderer.create(
      <ProductImage img="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRQDGQ55_znshhGDlxv5sDZz96tD3hxPc5j8CVWMKvJcw&s" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe("ProductImage", () => {
  test("Renders ProductImage with a the image of the product without crashing", () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
