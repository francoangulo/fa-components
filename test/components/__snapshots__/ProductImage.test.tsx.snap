// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Renders ProductImage with a custom image without crashing 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRQDGQ55_znshhGDlxv5sDZz96tD3hxPc5j8CVWMKvJcw&s"
/>
`;

exports[`ProductImage Renders ProductImage with a the image of the product without crashing 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
